### PR TITLE
Pass wasi invariant mode through env variable

### LINKED ISSUE #82567: [wasi] `Invariant.Tests` failing with `Failed to load ./icudt.dat`
On https://github.com/dotnet/runtime/pull/81649, [build](https://dev.azure.com/dnceng-public/public/_build/results?buildId=182683&view=logs&j=bc118297-27f2-5116-f3e6-9e173d056047&t=f24ab759-46e6-5bcd-4593-a21dd69c3f7c), and [Log](https://helixre107v0xdcypoyl9e7f.blob.core.windows.net/dotnet-runtime-refs-pull-81649-merge-fe982062cd664c9b8b/Invariant.Tests/1/console.ed14e776.log?helixlogtype=result):
```
[23:40:07] info: Failed to load ./icudt.dat
[23:40:07] info: Process wasmtime.exe exited with 3
[23:40:07] info: Error: failed to run main module `dotnet.wasm`
[23:40:07] info: Waiting to flush log messages with a timeout of 120 secs ..
[23:40:07] info: 
[23:40:07] info: Caused by:
[23:40:07] info:     0: failed to invoke command default
[23:40:07] info:     1: error while executing at wasm backtrace:
[23:40:07] info:            0: 0x26594e - <unknown>!fseek
[23:40:07] info:            1: 0x5b12 - <unknown>!load_icu_data
[23:40:07] info:            2: 0x5bd8 - <unknown>!mono_wasm_load_runtime
[23:40:07] info:            3: 0x60be - <unknown>!main
[23:40:07] info:            4: 0x262ec3 - <unknown>!__main_void
[23:40:07] info:            5: 0x58ee - <unknown>!_start
[23:40:07] info:            6: 0x276152 - <unknown>!_start.command_export
[23:40:07] info:        note: using the `WASMTIME_BACKTRACE_DETAILS=1` environment variable may show more debugging information
[23:40:07] info:     2: wasm trap: uninitialized element
[23:40:07] fail: Application has finished with exit code 3 but 0 was expected
```

cc @ilonatommy

### PR DESCRIPTION
**Pull request by mkhamoyan**
Fixes #82567

### CODE REVIEW

File: src/mono/wasi/runtime/driver.c [hunk 1]
```diff
@@ -369,11 +369,14 @@ mono_wasm_load_runtime (const char *unused, int debug_level)
 {
 	const char *interp_opts = "";
 
+    char* invariant_globalization = monoeg_g_getenv ("INVARIANT_GLOBALIZATION");
```
[the remark this answers is not in the document]
Comment (ilonatommy) [reply src/mono/wasi/runtime/driver.c:372]: Only `DOTNET_SYSTEM_GLOBALIZATION_INVARIANT` is public API as far as I know https://github.com/dotnet/runtime/blob/9a1c38e928abeba10d536f3c1f174de70a87627d/docs/design/features/globalization-invariant-mode.md?plain=1#L123
The idea sounds good.

File: src/mono/wasi/runtime/driver.c [hunk 2]
```diff
@@ -369,11 +369,14 @@ mono_wasm_load_runtime (const char *unused, int debug_level)
...
+    if (strcmp(invariant_globalization, "1") == 0)
+	   monoeg_g_setenv ("DOTNET_SYSTEM_GLOBALIZATION_INVARIANT", "true", 1);
+	else {
 #ifndef INVARIANT_GLOBALIZATION
```
Comment (ilonatommy) [inline src/mono/wasi/runtime/driver.c:376]: In the temporary solution we do not need the check for value of this directive, do we?

File: src/mono/wasi/runtime/driver.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (radical) [reply src/mono/wasi/runtime/driver.c:372]: I agree, using `DOTNET_SYSTEM_GLOBALIZATION_INVARIANT` will be preferable.

**Comment by radical** [comment]
Also, add `Invariant.Tests` to https://github.com/dotnet/runtime/blob/main/src/libraries/tests.proj#L538-L547 so that gets run, and the change here gets tested.

File: src/mono/wasi/runtime/driver.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (mkhamoyan) [reply src/mono/wasi/runtime/driver.c]: Updated.

File: src/mono/wasi/runtime/driver.c (hunk 2 above)
Comment (mkhamoyan) [reply src/mono/wasi/runtime/driver.c]: Refactored this part after using DOTNET_SYSTEM_GLOBALIZATION_INVARIANT.

**Comment by ilonatommy** [comment]
Failures in [`Build Libraries Test Run release mono_interpreter linux x64 Debug`](https://github.com/dotnet/runtime/pull/82707/checks?check_run_id=11683683974) not connected, e.g. [189037](https://dev.azure.com/dnceng-public/public/_build/results?buildId=189037&view=results) has them as well.

**Review by ilonatommy — APPROVED** [review]

**Pull request merged**